contracts/number/Math.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/utils/math/SignedMath.sol";
+
+/**
+ * @title Math
+ * @notice Library for additional math functions that are not included in the OpenZeppelin libraries.
+ */
+library Math {
+    /**
+     * @notice Divides `a` by `b`, rounding the result away from zero if there is a remainder
+     * @param a Dividend
+     * @param b Divisor
+     * @return Resulting quotient
+     */
+    function divOut(uint256 a, uint256 b) internal pure returns (uint256) {
+        if (a == 0 && b != 0) return 0;
```

Comment:
something really tricky we could do here would be to actually throw a real divide-by-zero error when we see `b==0` using this: https://docs.soliditylang.org/en/v0.8.17/control-structures.html#panic-via-assert-and-error-via-require

Then we can just use `Math.ceilDiv` directly from OZ:

```
error Panic(uint256 code);

function divOut(uint256 a, uint256 b) internal pure returns (uint256) {
    if (b == 0) Panic(0x12);
    return Math.ceilDiv(a, b);
}
```

Reply:
not 100% sure if this works, but if it does I think this is cleaner

Reply:
This doesn't work. If you like using `Math.ceilDiv` directly, though, then we can just revert with a custom error instead.

Reply:
Turns out, using `Math.ceilDiv` actually uses less gas somehow, so we decided to do this.